### ISSUE TITLE
Support for half float textures

### ISSUE THREAD

**Issue by marklundin** [stackgl/gl-fbo#11]
Like the title says, It would be really handy to add support for half float textures. This would be really beneficial for mobile devices

I'm not sure how the api would look, however the current `preferFloat`, `float` is a bit error prone and ambiguous if you consider `{ float:false, preferFloat:true }`. Could it be better to just pass int `gl.FLOAT` or `ext.HALF_FLOAT_OES` directly?

**Comment by mikolalysenko** [comment]
Hmm...  Maybe the way to deal with this is to have the precision be some string parameter?

Perhaps the user cold specify an argument like:

`{ float: '>16' }`  Need at least half float
`{ float: '32' }`  Need exactly float
`{ float:`>8`}` Use highest available precision

**Comment by marklundin** [comment]
Was also thinking about some bit flag, like {format: FLOAT | HALF_FLOAT | UNSIGNED_BYTE } so you can set the preferred order. Not sure if that makes sense though

**Comment by mikolalysenko** [comment]
Maybe..  Sometimes with float textures it is desirable to have them as an upgradeable option, since maybe you can use lower precision and get away with it.  Perhaps you could specify a list of precisions like:

`{ precision: [32, 16, 8] }`

**Comment by marklundin** [comment]
Yeah, that makes more sense to me, although I can't imagine a scenario wher you'd require `[ 8, 32, 16]` or something out of order.

**Comment by mikolalysenko** [comment]
I guess if you put 8 first, you'll always get it.  But there might be situations where maybe you'd prefer 32 vs 16 in different orders (based on memory restrictions).  I'm open to suggestions here on alternative ways to do this.

**Comment by neomantra** [comment]
I took a slightly different approach from the `precision` idea.  I added two options, `halfFloat` and `preferHalfFloat`.  So now we have:
- `float`: must be 32-bit
- `halfFloat`: must be 16-bit
- `preferFloat`: equivalent to `precision: [32, 16, 8]` in the ideas above
- `preferHalfFloat`: equivalent to `precision: [16, 32, 8]`

I also probe if these are actually renderable by the context.  It's a little extra work, but FBOs shouldn't be created very often.

EDIT: had mistyped `preferHalfFloat` as `preferFloat` in the bullets

**Comment by benwest** [comment]
yes! please merge this!!
